lib/new.js

```diff
@@ -221,102 +228,99 @@ const promptForName = (args) => {
+      { validator }
+    );
   };
 
   if (args.name) {
     try {
-      return success(validator(args.name));
+      return validator(args.name);
     } catch (err) {
       logger.error(err.message);
       return prompt();
     }
-  } else {
-    return prompt();
   }
-};
 
-/**
- * Returns a string to be used for the name property in package.json.
- * @name getPackageName
- */
-const getPackageName = (name) => {
-  let prefix;
-
-  switch (settings.template.name) {
-    case 'library':
-      prefix = 'lib';
-      break;
-    default:
-      prefix = 'spa';
-      break;
-  }
-
-  return `skyux-${prefix}-${name}`;
-};
+  return prompt();
+}
 
 /**
  * Returns an object representing the template config, derived
  * from constructor arguments.
  * @name getTemplateFromArgs
  */
-const getTemplateFromArgs = (args) => {
-  let name = 'default';
-  let url = 'https://github.com/blackbaud/skyux-template';
+function getTemplateFromArgs(args) {
+  const defaults = {
+    name: 'default',
+    url: 'https://github.com/blackbaud/skyux-template'
+  };
 
-  if (args.t) {
-    args.template = args.t;
+  let name = '';
+  if (args.template && typeof args.template === 'string') {
+    // Allow custom template with `--template`.
+    name = args.template;
+  } else if (args.t && typeof args.t === 'string') {
+    // The `-t` flag is an alias for `--template`.
+    name = args.t;
   }
 
-  // Checks if the flag is paired with a string.
-  // (If the flag is provided without the string, it will return `true`.)
-  if (typeof args.template === 'string') {
-    name = args.template;
-    if (args.template.indexOf(':') > -1) {
-      url = args.template;
-    } else {
-      url = `${url}-${name}`;
-    }
+  if (!name) {
+    return defaults;
   }
 
-  return Object.freeze({ name, url });
-};
+  const template = {
+    name
+  };
 
-const notify = () => {
-  logger.info('SPA %s created in directory %s', settings.spa, settings.name);
-  logger.info('Change into that directory and run "skyux serve" to begin.');
-  return Promise.resolve();
-};
+  if (name.indexOf(':') > -1) {
+    template.url = name;
+  } else {
+    template.url = `${defaults.url}-${name}`;
+  }
 
-/**
- * ENTRY POINT
- */
-module.exports = (args) => {
-  settings.template = getTemplateFromArgs(args);
-
-  return promptForName(args)
-    .then(() => promptForUrl(args))
-    .then(cloneRepo)
-    .then(cloneTemplate)
-    .then(getLatestVersions)
-    .then((v) => cleanupTemplate(v[0], v[1]))
-    .then(checkoutBranch)
-    .then(() => npmInstall(settings))
-    .then(notify)
-    .catch(logger.error);
-};
+  return template;
+}
+
+async function skyuxNew(args) {
+  try {
+    settings.template = getTemplateFromArgs(args);
+
+    const projectName = await promptForName(args);
+    settings.spa = projectName;
+    settings.name = getPackageName(projectName);
+    settings.path = path.join('.', settings.name);
+    settings.pathTmp = path.join(settings.path, 'tmp');
```

Comment:
I moved all the `settings` assignments to the entry point function to make them easier to locate in the future. This also makes the other functions more "pure" since the settings assignments are no longer created as side effects.